### ISSUE TITLE
Function.deps.json references .NET 6 instead of .NET 8, causing security vulnerabilities to be flagged

### ISSUE THREAD

**Issue by tomings** [Azure/azure-functions-durable-extension#2759]
We've got an isolated Azure function written in .NET 8, and some packages in the .azurefunctions folder are being flagged by some security software we use, specifically Azure.Identity and Microsoft.AspNetCore.Http.Features.

I've noticed that in the function.deps.json file, it references .NET 6 instead of .NET 8 and wondered if this could be the issue. How can we change this to reference .NET 8?

![Image1](https://github.com/Azure/azure-functions-dotnet-worker/assets/4939568/f76869fa-9e76-4549-aa56-1c74d84773eb)

![Image2](https://github.com/Azure/azure-functions-dotnet-worker/assets/4939568/b7fc5e5a-a98f-456c-aa7b-7ba3bf33b656)

![Image3](https://github.com/Azure/azure-functions-dotnet-worker/assets/4939568/386fe96a-a92e-4a32-a16f-2b3808980ace)

**Comment by davidmrdavid** [comment]
Just transferred this to the DF Extension repo, as these dependencies are being pulled due to our package.

@tomings: We've been notified of these security warnings and are working to update them. I'm looking to _start_ a hotfix deployment as early as next Monday. As soon as the deployment officially starts, we'll be able to provide you with a GitHub issue you can track to track the release progress. We'll keep you posted.

**Comment by tomings** [comment]
@davidmrdavid thank you for the update!

**Comment by facundoam** [comment]
Hi,

I have the same problem, I tried updating the DurableTask package to the latest rc version, but Azure.Identity.dll in .azurefunctions is 1.10.0 which is also vulnerable. 

Minimum required is 1.10.2+

<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <AzureFunctionsVersion>v4</AzureFunctionsVersion>
    <OutputType>Exe</OutputType>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <UserSecretsId>e0e5db94-dab7-4b5d-8451-55042a3554bf</UserSecretsId>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Azure.Identity" Version="1.10.4" />
    <PackageReference Include="Azure.Messaging.ServiceBus" Version="7.17.4" />
    <PackageReference Include="Azure.Storage.Blobs" Version="12.19.1" />
    <PackageReference Include="ceTe.DynamicPDF.CoreSuite.NET" Version="12.15.0" />
    <PackageReference Include="FluentValidation.AspNetCore" Version="11.3.0" />
    <PackageReference Include="Microsoft.ApplicationInsights.WorkerService" Version="2.22.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.21.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.ApplicationInsights" Version="1.2.0" />
    **<PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.DurableTask" Version="1.2.0-rc.1" />**
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.1.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.OpenApi" Version="1.5.1" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Timer" Version="4.3.0" />
    **<PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.17.2" />**
    <PackageReference Include="Microsoft.Extensions.Configuration.AzureAppConfiguration" Version="7.1.0" />
    <PackageReference Include="Microsoft.Extensions.Configuration.UserSecrets" Version="8.0.0" />
    <PackageReference Include="RestSharp" Version="110.2.0" />
  </ItemGroup>
  <ItemGroup>
    <None Update="host.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
    <None Update="local.settings.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <CopyToPublishDirectory>Never</CopyToPublishDirectory>
    </None>
  </ItemGroup>
  <ItemGroup>
    <Using Include="System.Threading.ExecutionContext" Alias="ExecutionContext" />
  </ItemGroup>
</Project>

![image](https://github.com/Azure/azure-functions-durable-extension/assets/14048333/11118a65-ed70-4de6-86a6-7b1e94386c90)

**Comment by facundoam** [comment]
Just saw this PR:

https://github.com/Azure/azure-functions-durable-extension/pull/2761

Thanks for all the updates!

**Comment by davidmrdavid** [comment]
Yep, that PR was just merged :) . We should be able to kickstart a release next week, though I have yet to create the tracking issue.

We have some known CI failures we have to tackle (just some dropped Azure DevOps tasks we need to work around), so I'm hoping those don't delay us much, if at all.

**Comment by tomings** [comment]
@davidmrdavid Not sure if this will be addressed by your fix, but we're also getting a security vulnerability flagged for the Microsoft.AspNetCore.Http.Features package. It's finding version 2.2.0, but we need 5.0.17. 

Strangely, this DLL doesn't appear anywhere in the bin or .azurefunctions folder. I can only see it referenced in the function.deps.json file.

**Comment by tomings** [comment]
Thanks for the link and the quick reply!

**Comment by davidmrdavid** [comment]
The new .NET isolated extension package is out: https://www.nuget.org/packages/Microsoft.Azure.Functions.Worker.Extensions.DurableTask/1.1.2

which contains a reference to the WebJobs extension, which contained the outdated `Azure.Identity` dependency: https://www.nuget.org/packages/Microsoft.Azure.WebJobs.Extensions.DurableTask/2.13.2

@tomings - can you please let me know if this addressed your vulnerability warnings?

**Comment by tomings** [comment]
@davidmrdavid It seems to have fixed the `Azure.Identity` issue, but `Microsoft.AspNetCore.Http.Features` is still being flagged as being on version 2.2.0 instead of 5.0.17. I tried removing the DurableTask package altogether, and it was no longer flagged, so it looks like it's something in this package.

**Comment by facundoam** [comment]
This fixed the vuln for us:

![image](https://github.com/Azure/azure-functions-durable-extension/assets/14048333/693c4a93-1ce4-42b8-a563-e66515e89c4e)


Thanks!

**Comment by davidmrdavid** [comment]
Thanks @tomings - I did a quick search on NuGet to reverse engineer where the dependency is coming from (if anyone has a better way to find these **nested** transitive dependencies, I'd love to learn it :-) ), and it's likely a result of our Kesterl dependency set to "[Microsoft.AspNetCore.Server.Kestrel](https://www.nuget.org/packages/Microsoft.AspNetCore.Server.Kestrel/) (>= 2.2.0)", which in turn brings this `"Http.Features" package at version `>= 2.2.0`.

I'll need to triple check if it's safe to update this dependency. I'll follow up.

**Comment by tomings** [comment]
@davidmrdavid I'm not sure what goes on under the hood, but we're using Veracode as a static scanning tool. I'm not at work right now to double check, but this looks like the same issue we're seeing: https://github.com/Azure/azure-sdk-for-net/issues/36411

**Comment by tomings** [comment]
@davidmrdavid Just wondering if there are any updates on this?

**Comment by unionthugface** [comment]
Veracode is flagging the function.deps.json file reference, and updating the NuGet packages does not update the references in the file. I also have no further reference to the Durable extensions, but problem persists.

**Comment by davidmrdavid** [comment]
Hi @unionthugface, @tomings - If the issue being flagged is Kestrel, then that should be only a static dependency. At runtime, the Durable Functions extension runs in the same process as the Azure Functions host, which loads up to date AspNetCore dependencies. You should be able to validate this at runtime by looking at the DLLs that get loaded in the Host process.

Unfortunately, those old dependencies cannot be removed without a major breaking change. The good news is that we're preparing a new major release right now that removes many such old dependencies: https://github.com/Azure/azure-functions-durable-extension/pull/2864 . So, until that release, and assuming we're talking about the Kestrel static dependency, we will need to flag it as a false alarm.

**Comment by aaronyiow97** [comment]
Hi, I am also facing the same issue with Azure.Identity and Microsoft.Identity.Client. I found out the root cause is coming from the Microsoft.Azure.Functions.Worker.Extensions.EventHubs library. I am on version 6.3.1 and this version is having a dependency of Microsoft.Extensions.Azure v1.7.3 and the Microsoft.Extensions.Azure is having a dependency on Azure.Identity v1.11.0 (which is flagged as vulnerable) and the Azure.Identity is also having a dependency on Microsoft.Identity.Client v4.60.1 (which also flagged as vulnerable).

![image](https://github.com/Azure/azure-functions-durable-extension/assets/60057174/0d206179-038b-49c6-9594-340b229e1734)

**Comment by srikanthkvl** [comment]
Problem still persists with "Microsoft.AspNetCore.Http.Features". It is still referencing old version 2.2.0 even after updating Microsoft.Azure.Functions.Worker.Extensions.DurableTask to 1.1.7 version

**Comment by torosent** [comment]
@srikanthkvl @unionthugface , is this problem still exists?
